src/cli.ts

```diff
@@ -0,0 +1,378 @@
+          ...args[i]
+            .split(",")
+            .map((f) => f.trim())
+            .filter(Boolean),
+        );
+        break;
+
+      case "--file-list":
+      case "-F": {
+        i++;
+        if (i >= args.length) {
+          throw new Error(`Missing value for ${arg}`);
+        }
+        // Read files from a file (one per line)
+        const filePath = path.resolve(args[i]);
+        if (!fs.existsSync(filePath)) {
+          throw new Error(`File not found: ${filePath}`);
+        }
+        const fileContent = fs.readFileSync(filePath, "utf-8");
+        options.files.push(
+          ...fileContent
+            .split("\n")
+            .map((f) => f.trim())
+            .filter(Boolean),
+        );
+        break;
+      }
+
+      case "--context":
+      case "-c": {
+        i++;
+        if (i >= args.length) {
+          throw new Error(`Missing value for ${arg}`);
+        }
+        const contextLines = Number.parseInt(args[i], 10);
+        if (Number.isNaN(contextLines) || contextLines < 0) {
+          throw new Error(`Invalid context lines value: ${args[i]}`);
+        }
+        options.contextLines = contextLines;
+        break;
+      }
+
+      case "--output":
+      case "-o":
+        i++;
+        if (i >= args.length) {
+          throw new Error(`Missing value for ${arg}`);
+        }
+        if (args[i] !== "markdown" && args[i] !== "unified") {
+          throw new Error(
+            `Invalid output format: ${args[i]}. Must be 'markdown' or 'unified'`,
+          );
+        }
+        options.output = args[i] as "markdown" | "unified";
+        break;
+
+      case "--help":
+      case "-h":
+        printHelp();
+        process.exit(0);
+        break;
+
+      case "--version":
+      case "-v":
+        printVersion();
+        process.exit(0);
+        break;
+
+      default:
+        // Unknown argument
+        throw new Error(`Unknown argument: ${arg}`);
+    }
+
+    i++;
+  }
+
+  return options;
+}
+
+/**
+ * Validate CLI options and throw if required options are missing.
+ * @param options CLI options to validate
+ */
+export function validateOptions(options: CliOptions): void {
+  if (!options.startCommit) {
+    throw new Error("Missing required argument: --start (-s)");
+  }
+  if (!options.endCommit) {
+    throw new Error("Missing required argument: --end (-e)");
+  }
+  if (options.files.length === 0) {
+    throw new Error(
+      "Missing required argument: --files (-f) or --file-list (-F)",
+    );
+  }
+}
+
+/**
+ * Print help message.
+ */
+export function printHelp(): void {
+  console.log(`
+pickdiff - Generate diffs between Git commits for specific files
+
+Usage:
+  pickdiff [options]
+
+Required Options:
+  -s, --start <commit>     Start commit (base commit)
+  -e, --end <commit>       End commit (target commit)
+  -f, --files <files>      Comma-separated list of files to diff
+  -F, --file-list <path>   Path to file containing list of files (one per line)
```

Comment:
Should we add a note in here somewhere that mentions that only one of `--files` or `--file-list` is required (not both).